### DISCUSSION TITLE
Evaluating items takes 15+ seconds at vendors

### DISCUSSION THREAD

**Discussion by tyson-dev** [isid0re/SoloLeveling#348] [General]
At Vendor my Classic Ladder Sorc (Blizzballer) takes very long for evaluating items ( 8+ secs each).

Example A4 Jamella Nightmare SCCL , this is exactly how they are in console, explanations in brackets:

InitPcBuyPotions
Minishopbot: scanning 58 items.
Sololeveling: evaluating 58 items. ( this takes like 8+ seconds before he goes next line again)
Minishopbot: scanning 58 items.
Sololeveling: evaluating 58 items. ( for a second time also taking up 8+ seconds before he goes finally...)
Ancient Armor. ( it just says "Ancient Armor" in console everytime after evaluating these 2 times like posted here, never actually buying one)
SoloLeveling: Inventory has used 7.5% of its total space

This is eating up huge amounts of time.

I'll check if this is also a thing in non-classic later...

Edit: i understand the "Ancient Armor" console now, this is stash. It says "ancient armor" and "Towerr shield" now as there is silks of the victor and sigon shield in stash.

**Comment by isid0re** [comment]
This is in classic? I think I have a solution I'll need to test it but the gist in some pseudocode is

If(townrepair) {
    Do second shopping;
}

As the second shopping is suppose to trigger after repairs. So you are right It is a waste of time if it doesnt go to repair.

Thanks.

**Reply by tyson-dev** [reply]
Hey Isi,

yeah this is my classic sorc, but i think the bigger Problem is that the evaluation itself takes so long, at least you're cutting it half if she doesn't do it 2 time :D Maybe worth thinking about getting rid off Shopping at all at some point, maybe with starting nightmare, stealth and ancipledge done then anyway i guess and for smoke she uses dropped items anyway (if you're NOT in classic ofc ).
